### PR TITLE
Update docker.yml and Dockerfile, fix ci

### PR DESCRIPTION
**Pull request by faddat**
Use github container registry

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,26 +1,12 @@
-FROM golang:1.14 as build
-
-WORKDIR /iavl
+# This docker image is designed to be used in CI for benchmarks and also by developers wanting an environment that always has the lastest rocksdb and cleveldb. 
+FROM faddat/archlinux
```
Comment (robert-zaremba) [inline Dockerfile:2]: how about using one of the official arch or alpine linux distros?

**Review by robert-zaremba — APPROVED** [review]
utACK

File: Dockerfile (hunk 1 above)
Comment (robert-zaremba) [reply Dockerfile:2]: ping

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -11,45 +12,30 @@ on:
...
+        name: Build and push
+        uses: docker/build-push-action@v2
         with:
-          file: ./Dockerfile
-          push: ${{ github.event_name != 'pull_request' }}
-          tags: ${{ steps.prep.outputs.tags }}
+          context: .
+          platforms: linux/amd64,linux/arm64
+          push: true
+          tags: ghcr.io/${{ github.repository_owner }}/iavl:latest
```
Comment (marbar3778) [inline .github/workflows/docker.yml:41]: love this!! we should migrate everything we have to this

File: Dockerfile (hunk 1 above)
Comment (faddat) [reply Dockerfile:2]: hey sure thing

File: .github/workflows/docker.yml (hunk 2 above)
Comment (faddat) [reply .github/workflows/docker.yml:41]: so, ok-- this is a package deal with the above docker image, at least in my experience the multiarch stuff.  Let me see what I can do.

**Comment by marbar3778** [comment]
we removed this docker file

**Pull request closed without merging**